### PR TITLE
[CHANGE] Kv.keys() returns an iterator 

### PR DESCRIPTION
**Pull request by aricart**

### CODE REVIEW

**Review by kozlovic — APPROVED** [review]
LGTM, but note about check of key with wildcards in delete/purge.

File: nats-base-client/kv.ts [hunk 1]
```diff
@@ -346,28 +349,22 @@ export class Bucket implements KV, KvRemove {
     return this._deleteOrPurge(k, "DEL");
   }
 
-  async _deleteOrPurge(k: string, op: "DEL" | "PURGE") {
+  async _deleteOrPurge(k: string, op: "DEL" | "PURGE"): Promise<void> {
```
[the remark this answers is not in the document]
Comment (aricart) [reply nats-base-client/kv.ts:352]: I implemented before the server was doing additional support for KV, let me do a test, perhaps it is doing so already

[the remark this answers is not in the document]
Comment (aricart) [reply nats-base-client/kv.ts:352]: @kozlovic just verified, the server has implemented this functionality. I'll submit a new PR removing the hand crafted deletes

[the remark this answers is not in the document]
Comment (aricart) [reply nats-base-client/kv.ts:352]: @kozlovic and @ripienaar so the interesting part is that if you do a delete on `a.*` it does the deletion, and registers a tombstone on `a.*` which is not great, and worse, the publish had a subject of `a.*` with the kv prefix, which is illegal, but goes through...

[the remark this answers is not in the document]
Comment (aricart) [reply nats-base-client/kv.ts:352]: this means we have to do it by hand....

**Pull request merged**